components/eamxx/src/physics/gw/impl/gw_gwd_compute_tendencies_from_stress_divergence_impl.hpp

```diff
@@ -47,22 +46,16 @@ void Functions<S,D>::gwd_compute_tendencies_from_stress_divergence(
 {
   const Real ptaper = do_taper ? std::cos(lat) : 1.;
 
-  // Force tau at the top of the model to zero, if requested.
-  if (init.tau_0_ubc) {
-    Kokkos::parallel_for(
-      Kokkos::TeamVectorRange(team, 0, tau.extent(0)), [&] (const int i) {
-      tau(i,0) = 0.;
-    });
-  }
-
-  team.team_barrier();
-
   // Loop waves
   Kokkos::parallel_for(
-    Kokkos::TeamVectorRange(team, -ngwv, ngwv+1), [&] (const int l) {
+    Kokkos::TeamVectorRange(team, -pgwv, pgwv+1), [&] (const int l) {
     //  Accumulate the mean wind tendency over wavenumber.
-    int nl_idx = l + ngwv; // 0-based idx for -ngwv:ngwv arrays
-    int pl_idx = nl_idx + (pgwv - ngwv); // 0-based idx -pgwv:pgwv arrays
+    int pl_idx = l + pgwv; // 0-based idx for -pgwv:pgwv arrays
+
+    // Force tau at the top of the model to zero, if requested.
```

Comment:
I _think_ this will cause compute-sanitizer to issue an error for the racecheck condition. I'm not sure, I don't know how it handles concurrent writes with the same compite-time number, maybe it recognizes that it's benign?